### DISCUSSION TITLE
Is it possible to save and reload more than one gameplay?

### DISCUSSION THREAD

**Discussion by stephan75** [flareteam/flare-game#957] [Q&A]
Hello,
when trying version 1.14 from https://flarerpg.org/download/ ...
is it correct that a user can save and load only ONE single gameplay slot?
And that savegame is automatically overwritten on exit the game?
Or did I miss something? But I found no hint about using more savegane slots on the github repo or elsewhere ...

Has there ever been any considerations for more savegame slots?

**Comment by dorkster** [comment]
Yes, there is only a single save slot per created character.

By default, the game is saved when:
- exiting the game
- changing maps
- before showing a cutscene
- when storing or removing items from the stash

This save behavior can be altered by mod authors with [some properties](https://github.com/flareteam/flare-engine/wiki/Attribute-Reference#enginesettings-misc) in engine/misc.txt:

**save\_onload** | `bool` | Save the game upon changing maps.

**save\_onexit** | `bool` | Save the game upon quitting to the title screen or desktop.

**save\_pos\_onexit** | `bool` | If the game gets saved on exiting, store the player's current position instead of the map spawn position.

**save\_oncutscene** | `bool` | Saves the game when triggering any cutscene via an Event.

**save\_onstash** | `[bool, "private", "shared"]` | Saves the game when changing the contents of a stash. The default is true (i.e. save when using both stash types). Use caution with the values "private" and false, since not saving shared stashes exposes an item duplication exploit.

**save\_anywhere** | `bool` | Enables saving the game with a button in the pause menu

**Reply by dorkster** [reply]
As for why we don't support multiple saves per character, I can't say as I didn't make that decision.

I imagine the purpose is to slightly discourage save-scumming, but that's just a guess. Any comments @clintbellanger?

**Comment by clintbellanger** [comment]
Most likely I only did auto-save because adding UI for manual saving is more work. 

Possibly there was a shifting philosophy at the time? Diablo was one of the first games I can remember to only auto-save instead of manual-save. Easy to lose progress on older games where there's only manual save.

In Flare maps mostly reset when you re-enter them, and our save files don't store live map states, so there's slightly less reason to support a "save anywhere" system. Instead we usually save when transitioning between maps so the player will respawn at the latest map entrance.

Save-scumming is fine in a single player open source game though, we do want players to go in and mess with their plain-text save files.

There is some element of "decisions matter" when saving is automatic, I happen to enjoy this but it's not strictly necessary for the engine.

**Reply by dorkster** [reply]
Thanks for the insight!

>There is some element of "decisions matter" when saving is automatic, I happen to enjoy this but it's not strictly necessary for the engine.

I'm also a fan of this style. But to expand on it, I think there are ways to go in the other direction without having the player manage save files. I think irreversible player choices can be split into two categories:

- Mechanical; e.g. stat & power point allocation
- Story; quest progress and general NPC interactions

It'd be possible to reset both of these in-game. As an example, there could be a special item that the player could trade in to:

- Respec their point allocations
- Reset and allow replaying a side-quest
- Restore an NPC's disposition to a neutral state
